march_rqt_gait_generator/src/march_rqt_gait_generator/joint_table_controller.py

```diff
@@ -0,0 +1,44 @@
+import math
+
+from python_qt_binding.QtWidgets import QTableWidgetItem
+
+from .joint_setting_spin_box_delegate import JointSettingSpinBoxDelegate
+from .model.modifiable_setpoint import ModifiableSetpoint
+
+
+class JointTableController(object):
```

Comment:
I don't know exactly if its possible in this situation, you can inherit from [`QAbstractTableModel`](https://doc.qt.io/qtforpython/PySide2/QtCore/QAbstractTableModel.html) to implement the `JointTable` as a model for the tables. You can take a look at [`march_rqt_note_taker.EntryModel`](https://github.com/project-march/monitor/blob/develop/march_rqt_note_taker/src/march_rqt_note_taker/entry_model.py). It is better supported by Qt and probably has a better performance.

Reply:
Hmm this is a little complicated so Imma make this a separate issue, okay?

Reply:
Nice :+1: